src/Features/LanguageServer/Protocol/LanguageServerProtocol.cs

```diff
@@ -54,235 +53,7 @@ public LanguageServerProtocol([ImportMany] IEnumerable<Lazy<IRequestHandler, IRe
             var handler = (IRequestHandler<RequestType, ResponseType>?)_requestHandlers[methodName]?.Value;
             Contract.ThrowIfNull(handler, string.Format("Request handler not found for method {0}", methodName));
 
-            return handler.HandleRequestAsync(solution, request, clientCapabilities, cancellationToken);
+            return handler.HandleRequestAsync(solution, request, clientCapabilities, clientName, cancellationToken);
         }
-
-        /// <summary>
-        /// Answers an execute workspace command request by handling the specified command name.
-        /// https://microsoft.github.io/language-server-protocol/specification#workspace_executeCommand
-        /// </summary>
-        /// <param name="solution">the solution relevant to the workspace.</param>
-        /// <param name="request">the request command name and arguments.</param>
-        /// <param name="clientCapabilities">the client capabilities for the request.</param>
-        /// <param name="cancellationToken">a cancellation token.</param>
-        /// <returns>any or null.</returns>
-        public Task<object> ExecuteWorkspaceCommandAsync(Solution solution, LSP.ExecuteCommandParams request, LSP.ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
```

Comment:
This was all just dead and it wasn't worth updating now?

Reply:
Not dead, but a layer of abstraction I felt was becoming more of annoyance than helpful